File: backend/src/main/java/com/clothing/service/CategoryService.java
package com.clothing.service;

import com.clothing.dto.request.CategoryUpsertRequest;
import com.clothing.dto.response.CategoryResponse;
import com.clothing.dto.response.PageResponse;

public interface CategoryService {

    CategoryResponse create(CategoryUpsertRequest request);

    CategoryResponse update(Long id, CategoryUpsertRequest request);

    void delete(Long id);

    CategoryResponse getById(Long id);

    PageResponse<CategoryResponse> getAll(int page, int size, String sortBy, String direction);
}

File: clothing-ios/clothing/Services/CategoryService.swift
import Foundation

final class CategoryService {
    static let shared = CategoryService()
    private init() {}

    func getCategories(page: Int = 0, size: Int = 20) async throws -> [Category] {
        let data = try await APIClient.shared.request(
            path: "/api/categories",
            queryItems: [
                URLQueryItem(name: "page", value: String(page)),
                URLQueryItem(name: "size", value: String(size)),
                URLQueryItem(name: "sortBy", value: "id"),
                URLQueryItem(name: "direction", value: "asc")
            ]
        )
        let response = try APIClient.shared.decodeResponse(PageResponse<Category>.self, from: data)
        return response.content ?? []
    }
}
